### ISSUE TITLE
Evidential models

### ISSUE THREAD

**Issue by Cortys** [Cortys/SEP-Probly-WS2526#25]
This epic introduces evidential training functionality to the package, enabling evidential models to be easily trained. 
Evidential Deep Learning predicts the $\boldsymbol{\alpha}$ parameters of a Dirichlet distribution (in the case of classification) offering direct uncertainty representation and quantification without requiring multiple models or sampling.

---

## Motivation and Context
Evidential Deep Learning (EDL), as introduced by [Sensoy et al. (2018)](https://arxiv.org/abs/1806.01768), provides a framework where networks output evidence parameters of a Dirichlet distribution, capturing both the prediction and its uncertainty.

Integrating evidential training into our framework will allow users to easily train their models without having to think about the details of the training procedure.

---

## Goals

- Implement a unified evidential training function with, at minimum, a common and torch implementation.
- Optionally, include flax.
- Allow users to choose their loss functions from the ones introduced by [Sensoy et al. (2018)](https://arxiv.org/abs/1806.01768).
- Implement other types of evidential models (see [Ulmer et al., 2023](https://arxiv.org/abs/2110.03051)):
  - Prior networks (Malinin & Gales, 2018)
  - Evidential Deep Learning (Sensoy et al., 2018)
  - Information Robust Dirichlet Networks (Tsiligkaridis, 2019)
  - Posterior Networks (Charpentier et al., 2020)
  - Deep Evidential Regression (Amini et al., 2020)
  - Regression Prior Network (Malinin et al., 2020)
  - Natural Posterior Network (Charpentier et al., 2022)
- Allow users to specify common training hyper-parameters (number of epochs, learning rate, etc.)
- Provide sensible defaults aligned with the original papers.
- Provide example notebooks and documentation illustrating usage, loss configuration, and performance interpretation.

---

## Constraints

- Implementation must use lazy dispatching. Have a look at the transformations module for an example.
- Use the existing losses for the evidential train module. 
- Default hyper-parameters and loss weights should match recommendations from the Evidential Deep Learning paper. 
- Follow project standards for documentation, testing, and reproducibility.  

---

## References

- [Sensoy, M., Kaplan, L., & Kandemir, M. (2018). Evidential Deep Learning to Quantify Classification Uncertainty. NeurIPS 2018.](https://arxiv.org/abs/1806.01768) 
- [Ulmer, D., Hardmeier, C., & Frellsen, J. (2023). Prior and Posterior Networks: A Survey on Evidential Deep Learning Methods For Uncertainty Estimation. TMLR 2023.](https://arxiv.org/abs/2110.03051)

---
